src/etc/inc/interfaces.inc

```diff
@@ -3035,11 +3035,21 @@ function kill_dhcp6client_process($interface) {
  	}
 
  	$i = 0;
+ 	
+ 	$pid = find_dhcp6c_process($interface);
+	
+	if($pid > 0){
+		mwexec("/usr/bin/logger -t mwtag 'found dhcp6 process {$pid}'"); // debug info only
+	}
+	else {
+		mwexec("/usr/bin/logger -t mwtag 'no dhcp6c process to kill'"); // debug info only
+		unset ($i);
+		return;
+	}
+	
  	while ((($pid = find_dhcp6c_process($interface)) != 0) && ($i < 3)) {
- 		/* 3rd time make it die for sure */
- 		$sig = ($i == 2 ? SIGKILL : SIGTERM);
- 		posix_kill($pid, $sig);
- 		sleep(1);
+ 		mwexec("kill -9 $pid");
```

Comment:
Just to add, the process ID was the same, so it was not another process being called. Although thinking on that, maybe a second dhcp6c was launched by rtsold ( that's what the other bits of this patch fix ) and that was killed instead of the first process. I'll hammer it again this evening and see if I can definitely prove it one way or the other,